File: exercises/day10-TimePortal/scala/src/main/scala/timeportal/PortalNetwork.scala
package timeportal

import scala.collection.mutable.ListBuffer

class PortalNetwork(val networkName: String):
  private val portals: ListBuffer[TimePortal] = ListBuffer.empty

  def addPortal(portal: TimePortal): Unit = portals += portal

  def getTotalEnergy: Int =
    var total = 0
    for i <- portals.indices do
      total = total + portals(i).energyLevel
    total

  def findActivePortalWithEnergy(minimumEnergy: Int): Option[TimePortal] =
    for i <- portals.indices do
      if portals(i).isActive && portals(i).energyLevel >= minimumEnergy then
        return Some(portals(i))
    None

  def getPortals: List[TimePortal] = portals.toList

File: exercises/day10-TimePortal/kotlin/src/main/kotlin/timeportal/PortalNetwork.kt
package timeportal

class PortalNetwork(val networkName: String) {
    private val portals: MutableList<TimePortal> = mutableListOf()

    fun addPortal(portal: TimePortal) {
        portals.add(portal)
    }

    fun getTotalEnergy(): Int {
        var total = 0
        for (i in portals.indices) {
            total += portals[i].energyLevel
        }
        return total
    }

    fun findActivePortalWithEnergy(minimumEnergy: Int): TimePortal? {
        for (i in portals.indices) {
            if (portals[i].isActive && portals[i].energyLevel >= minimumEnergy) {
                return portals[i]
            }
        }
        return null
    }

    fun getPortals(): List<TimePortal> = portals
}
